api/src/wfl/module/aou.clj

```diff
@@ -47,17 +45,31 @@
    :disk_size                        100
    :preemptible_tries                3})
 
-(defn map-aou-environment
-  "Map AOU-ENV to environment for inputs preparation."
-  [aou-env]
-  ({:aou-dev "dev" :aou-prod "prod"} aou-env))
+(def ^:private known-cromwells
+  ["https://cromwell-gotc-auth.gotc-dev.broadinstitute.org"
+   "https://cromwell-aou.gotc-prod.broadinstitute.org"])
 
-(defn env-inputs
-  "Array inputs for ENVIRONMENT that do not depend on the input file.
-  Environment here is just a default, input file may specify override."
-  [environment]
-  {:vault_token_path (get-in env/stuff [environment :vault_token_path])
-   :environment      (map-aou-environment environment)})
+(def ^:private inputs+options
+  [{:environment "dev"
+    :vault_token_path "gs://broad-dsp-gotc-arrays-dev-tokens/arrayswdl.token"}
+   {:environment "prod"
+    :vault_token_path "gs://broad-dsp-gotc-arrays-prod-tokens/arrayswdl.token"}])
```

Comment:
Are these the pipeline environments?
Do we really want to hard code these? Might there be more? Might they change?

Reply:
Sorry for the confusion, these are specific to `aou` module, as the Arrays pipeline takes an `environment` String as one of the inputs and use it to decide which database (one of dev, staging and prod) to write the cloud metrics during the pipeline execution. We had this block in `environments.clj` for so long and caused a lot of confusion and trouble, thus I moved it into the module itself to narrow the scope of what does `environment` really mean. This can still be overwritten by the incoming notifications as we have to support Mercury "staging" tests.

I was hoping after this PR, WFL will mostly pass (cromwell/executor) URLs across the codebase and stop using `environment` as a term to represent "processing envs" such as `wgs-dev` or `aou-prod` as they are hard-coded in the modules. The only `environment` WFL should care about maybe `debug` which supports local development then.

Reply:
The "debug" environment was a hack. Would be nice to discard it, but by all means keep it if it solves some problem.

Reply:
We could add those to the process environment.
(-In for a penny; in for a pound!-)

Reply:
We could, a few string variables should be fine. Don't want to serialize/de-serialize a bunch of maps though from environment variables (process environment).